babel.config.js

```diff
@@ -25,99 +25,394 @@
+const argv = require('minimist')(process.argv.slice(2));
+const experimentsConfig = require('./build-system/global-configs/experiments-config.json');
+const experimentsConstantBackup = require('./build-system/global-configs/experiments-const.json');
+const fs = require('fs');
 
-const isClosureCompiler =
-  argv._.includes('dist') ||
-  argv._.includes('check-types') ||
-  (argv._.length == 0 && argv.compiled);
-const {esm} = argv;
+/**
+ * Default options for transform-react-jsx. Used by pre-closure and dep-check.
+ */
+const defaultJsxOpts = {
+  pragma: 'Preact.createElement',
+  pragmaFrag: 'Preact.Fragment',
+  useSpread: true,
+};
+
+/**
+ * Default preset-env plugin. Used by dep-check and tests.
+ */
+const defaultPresetEnvPlugin = [
+  '@babel/preset-env',
+  {
+    bugfixes: true,
+    modules: 'commonjs',
+    loose: true,
+    targets: {'browsers': ['Last 2 versions']},
+  },
+];
+
+/**
+ * Default filter-imports plugin. Used by pre-closure.
+ */
+const defaultFilterImportsPlugin = [
+  'filter-imports',
+  {
+    imports: {
+      './polyfills/document-contains': ['installDocContains'],
+      './polyfills/domtokenlist': ['installDOMTokenList'],
+      './polyfills/fetch': ['installFetch'],
+      './polyfills/math-sign': ['installMathSign'],
+      './polyfills/object-assign': ['installObjectAssign'],
+      './polyfills/object-values': ['installObjectValues'],
+      './polyfills/promise': ['installPromise'],
+      './polyfills/array-includes': ['installArrayIncludes'],
+      '../third_party/css-escape/css-escape': ['cssEscape'],
+      '../build/ampshared.css': ['cssText', 'ampSharedCss'],
+      '../build/ampdoc.css': ['cssText', 'ampDocCss'],
+    },
+  },
+];
+
+/**
+ * Default istanbul plugin. Used by tests.
+ */
+const defaultInstanbulPlugin = [
+  'istanbul',
+  {
+    exclude: [
+      'ads/**/*.js',
+      'build-system/**/*.js',
+      'extensions/**/test/**/*.js',
+      'third_party/**/*.js',
+      'test/**/*.js',
+      'testing/**/*.js',
+    ],
+  },
+];
+
+/**
+ * Gets relative paths to all the devDependencies defined in package.json.
+ *
+ * @return {!Array<string>}
+ */
+function devDependencies() {
+  const file = fs.readFileSync('package.json', 'utf8');
+  const packageJson = JSON.parse(file);
+  const devDependencies = Object.keys(packageJson['devDependencies']);
+  return devDependencies.map((p) => `./node_modules/${p}`);
+}
+
+/**
+ * Ignore devDependencies except for 'chai-as-promised' which contains ES6 code.
+ * ES6 code is fine for most test environments, but not for integration tests
+ * running on SauceLabs since some older browsers need ES5.
+ */
+const ignoredGlobalModules = devDependencies().filter(
+  (dep) => dep.indexOf('chai-as-promised') === -1
+);
+
+/**
+ * Resolves the full path of a local plugin with the given name.
+ *
+ * @param {string} name
+ * @return {string}
+ */
+function localPlugin(name) {
+  return require.resolve(`./build-system/babel-plugins/babel-plugin-${name}`);
```

Comment:
Not sure this abstraction provides value, but a different PR would be a smart idea to discuss and potentially address.

Reply:
This was added to shorten each line and prevent Prettier from making the code extra verbose. With this refactor, it no longer makes a huge difference.

Removed.